technologies/job/aws-lambda/functions/instanceActions.js

```diff
@@ -0,0 +1,184 @@
+const { Response, JobStatus, Log, Stream } = require('@saagie/sdk');
+const AWS = require('aws-sdk');
+
+/**
+ * Logic to start the external job instance.
+ * @param {Object} params
+ * @param {Object} params.job - Contains job data including featuresValues.
+ * @param {Object} params.instance - Contains instance data.
+ */
+exports.start = async ({ job, instance }) => {
+  try {
+    console.log('START INSTANCE:', instance);
+    AWS.config.update({credentials: { accessKeyId : job.featuresValues.endpoint.aws_access_key_id, secretAccessKey:  job.featuresValues.endpoint.aws_secret_access_key}});
+    AWS.config.update({region: job.featuresValues.endpoint.region});
+
+    const lambda = new AWS.Lambda({apiVersion: '2015-03-31'});
+
+    // Start all trigger/eventsource
+    var dataList = job.featuresValues.functions.sourceId.map(
+    (value) =>
+      lambda.updateEventSourceMapping({
+        Enabled: true, 
+        FunctionName: job.featuresValues.functions.id,
+        UUID: value
+      }).promise().then((data) => {return data.State; })
+    );
+    dataList=await Promise.all(dataList);
```

Comment:
```suggestion
    const dataList = job.featuresValues.functions.sourceId.map(
    (value) =>
      lambda.updateEventSourceMapping({
        Enabled: true, 
        FunctionName: job.featuresValues.functions.id,
        UUID: value
      }).promise().then((data) => {return data.State; })
    );
    
    await Promise.all(dataList);
```

:warning: **Important**: Here, I remove the `dataList` assignment on the last line, because it is not re-used after. So I'm questioning myself on the `}).promise().then((data) => {return data.State; })` line, is it really useful to return `data.State`?

:bulb: **Info**: It's better to use `const` (immutable), or at least `let` (mutable) over `var` for variable declaration. `var` should not be used anymore in any modern JavaScript file even if it works :wink:

Reply:
It was for debugging purpose, I am pushing a new commit on this.